presto-main/src/main/java/com/facebook/presto/operator/DriverAbortSignal.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DriverAbortSignal
+{
+    private final AtomicBoolean aborted = new AtomicBoolean();
```

Comment:
@haozhun had a benchmark comparing volatile and atomic boolean. If I recall correctly, volatile with padding (64 bytes before and 64 bytes after) has similar performance with atomic boolean. The one without won't.